src/commands.ts

```diff
@@ -90,27 +95,46 @@ export class Commands {
 			return;
 		}
 		const encryptDDO = args[2] === "false" ? false : true;
-		try {
-			// add some more checks
-			const urlAssetId = await createAsset(
-				asset.nft.name,
-				asset.nft.symbol,
-				this.signer,
-				asset.services[0].files,
-				asset,
-				this.providerUrl,
-				this.config,
-				this.aquarius,
-				1,
-				this.macOsProviderUrl,
-				encryptDDO
-			);
-			console.log("Asset published. ID:  " + urlAssetId);
-		} catch (e) {
-			console.error("Error when publishing dataset from file: " + args[1]);
-			console.error(e);
-			return;
+		let id: string
+		switch (asset.version) {
```

Comment:
wouldn´t be easier to check if is v5 and then do something different, otherwise just do as before? Not really sure why a need a switch/case for only 2 options?

Reply:
the thing is that, IMO,  you're doing a lot of refactoring for introducing a small variation.. This goes for the CLI and the Nodes.. So it's hard to even review it, because there are many file changes